src/Internal/Helpers.purs

```diff
@@ -232,23 +234,41 @@ showWithParens
   -> String
 showWithParens ctorName x = "(" <> ctorName <> " (" <> show x <> "))"
 
--- | A wrapper around the `dictionary` function from `Aeson.Encode`
--- | that uses `encodeAeson` for encoding keys and values.
+-- | If `k` is encoded as string, `encodeMap` encodes `Map` as `Object`,
+-- | else as an `Array`` of `Aeson /\ Aeson` pairs
```

Comment:
Does not this change behavior? If so it may break something?

Reply:
No, it does not. It is exactly the same as previous one (take a look how `dictionary` in helpers is implemented). 

From my point of view this is poor implementation, and it is still here exactly for compatibility reasons.